File: platforms/ios/foundation/Sources/NativeblocksFoundation/block/Spacer.swift
import NativeblocksCompiler
import NativeblocksRuntime
import SwiftUI

@Block(
    name: "Spacer",
    keyType: "nativeblocks/spacer",
    description: "Empty space; size it by attaching width/height modifiers.",
    version: 1,
    versionName: "1"
)
struct Spacer: View {
    var blockContext: BlockContext? = nil

    var body: some View {
        SwiftUI.Spacer(minLength: 0)
            .modifier(blockContext?.modifier ?? .none)
    }
}

File: platforms/android/foundation/foundation/src/main/java/io/nativeblocks/foundation/block/Spacer.kt
package io.nativeblocks.foundation.block

import androidx.compose.runtime.Composable
import io.nativeblocks.compiler.type.Block
import io.nativeblocks.runtime.api.provider.block.BlockContext
import androidx.compose.foundation.layout.Spacer as ComposeSpacer

@Block(
    keyType = "nativeblocks/spacer",
    name = "Spacer",
    description = "Empty space; size it by attaching width/height/weight modifiers.",
    version = 1,
    versionName = "1",
)
@Composable
internal fun Spacer(blockContext: BlockContext) {
    ComposeSpacer(modifier = blockContext.modifier)
}
